In [21]:
import pandas as pd 
import numpy as np
import plotly.express as px


In [22]:
df = pd.read_csv(r'mkt_data.csv')
df

Unnamed: 0   Income  Kidhome  Teenhome  Recency  MntWines  MntFruits  \
0              0  58138.0        0         0       58       635         88   
1              1  46344.0        1         1       38        11          1   
2              2  71613.0        0         0       26       426         49   
3              3  26646.0        1         0       26        11          4   
4              4  58293.0        1         0       94       173         43   
...          ...      ...      ...       ...      ...       ...        ...   
2200        2200  61223.0        0         1       46       709         43   
2201        2201  64014.0        2         1       56       406          0   
2202        2202  56981.0        0         0       91       908         48   
2203        2203  69245.0        0         1        8       428         30   
2204        2204  52869.0        1         1       40        84          3   

      MntMeatProducts  MntFishProducts  MntSweetProducts  ...  \
0                 546              172                88  ...   
1                   6                2                 1  ...   
2                 127              111                21  ...   
3                  20               10                 3  ...   
4                 118               46                27  ...   
...               ...              ...               ...  ...   
2200              182               42               118  ...   
2201               30                0                 0  ...   
2202              217               32                12  ...   
2203              214               80                30  ...   
2204               61                2                 1  ...   

      education_Graduation  education_Master  education_PhD  MntTotal  \
0                      3.0               NaN            NaN      1529   
1                      3.0               NaN            NaN        21   
2                      3.0               NaN            NaN       734   
3                      3.0               NaN            NaN        48   
4                      NaN               NaN            5.0       407   
...                    ...               ...            ...       ...   
2200                   3.0               NaN            NaN      1094   
2201                   NaN               NaN            5.0       436   
2202                   3.0               NaN            NaN      1217   
2203                   NaN               4.0            NaN       782   
2204                   NaN               NaN            5.0       151   

      MntRegularProds  AcceptedCmpOverall  marital_status  education_level  \
0                1441                   0          Single       Graduation   
1                  15                   0          Single       Graduation   
2                 692                   0        Together       Graduation   
3                  43                   0        Together       Graduation   
4                 392                   0         Married              PhD   
...               ...                 ...             ...              ...   
2200              847                   0         Married       Graduation   
2201              428                   1        Together              PhD   
2202             1193                   1        Divorced       Graduation   
2203              721                   0        Together           Master   
2204              130                   0         Married              PhD   

      kids  expenses  
0        0      1529  
1        2        21  
2        0       734  
3        1        48  
4        1       407  
...    ...       ...  
2200     1      1094  
2201     3       436  
2202     0      1217  
2203     1       782  
2204     2       151  

[2205 rows x 44 columns]

### Quantos dados temos? Linhas e colunas 


In [23]:
linhas,colunas = df.shape
print(f'Linhas: {linhas}, Colunas: {colunas}')

Linhas: 2205, Colunas: 44


 São 44 colunas e  2205 linhas 

### Quais são as colunas numéricas?


In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2205 entries, 0 to 2204
Data columns (total 44 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            2205 non-null   int64  
 1   Income                2205 non-null   float64
 2   Kidhome               2205 non-null   int64  
 3   Teenhome              2205 non-null   int64  
 4   Recency               2205 non-null   int64  
 5   MntWines              2205 non-null   int64  
 6   MntFruits             2205 non-null   int64  
 7   MntMeatProducts       2205 non-null   int64  
 8   MntFishProducts       2205 non-null   int64  
 9   MntSweetProducts      2205 non-null   int64  
 10  MntGoldProds          2205 non-null   int64  
 11  NumDealsPurchases     2205 non-null   int64  
 12  NumWebPurchases       2205 non-null   int64  
 13  NumCatalogPurchases   2205 non-null   int64  
 14  NumStorePurchases     2205 non-null   int64  
 15  NumWebVisitsMonth    

In [25]:
numerics = ['int16', 'int32', 'int64', 'float16', 'float32', 'float64']
lista_colunas_numericas = [coluna for coluna in df if df[coluna].dtype in numerics ]

lista_colunas_numericas

['Unnamed: 0',
 'Income',
 'Kidhome',
 'Teenhome',
 'Recency',
 'MntWines',
 'MntFruits',
 'MntMeatProducts',
 'MntFishProducts',
 'MntSweetProducts',
 'MntGoldProds',
 'NumDealsPurchases',
 'NumWebPurchases',
 'NumCatalogPurchases',
 'NumStorePurchases',
 'NumWebVisitsMonth',
 'AcceptedCmp3',
 'AcceptedCmp4',
 'AcceptedCmp5',
 'AcceptedCmp1',
 'AcceptedCmp2',
 'Complain',
 'Z_CostContact',
 'Z_Revenue',
 'Response',
 'Age',
 'Customer_Days',
 'marital_Divorced',
 'marital_Married',
 'marital_Single',
 'marital_Together',
 'marital_Widow',
 'education_2n Cycle',
 'education_Basic',
 'education_Graduation',
 'education_Master',
 'education_PhD',
 'MntTotal',
 'MntRegularProds',
 'AcceptedCmpOverall',
 'kids',
 'expenses']

No presente Conjuntos de dados , onde 44 e o número total de colunas, a grande maioria dessas colunas é do tipo numérico, somando 42 colunas numéricas('int64','float64') e apenas 2 colunas categórica('Object')

### Temos duplicados na nossa base? Se tivermos, retire-os 



In [26]:
df.duplicated().sum()

0

 Sem valores duplicados! Porém existe uma coluna de Index que nao é necessária para o DF 

In [27]:
df = df.drop("Unnamed: 0", axis=1)
df.duplicated().sum()

184

E ao retirar essa coluna verificamos que existe um total de 184 valores duplicados


In [28]:
df.drop_duplicates()

Income  Kidhome  Teenhome  Recency  MntWines  MntFruits  \
0     58138.0        0         0       58       635         88   
1     46344.0        1         1       38        11          1   
2     71613.0        0         0       26       426         49   
3     26646.0        1         0       26        11          4   
4     58293.0        1         0       94       173         43   
...       ...      ...       ...      ...       ...        ...   
2198  26816.0        0         0       50         5          1   
2200  61223.0        0         1       46       709         43   
2202  56981.0        0         0       91       908         48   
2203  69245.0        0         1        8       428         30   
2204  52869.0        1         1       40        84          3   

      MntMeatProducts  MntFishProducts  MntSweetProducts  MntGoldProds  ...  \
0                 546              172                88            88  ...   
1                   6                2                 1             6  ...   
2                 127              111                21            42  ...   
3                  20               10                 3             5  ...   
4                 118               46                27            15  ...   
...               ...              ...               ...           ...  ...   
2198                6                3                 4             3  ...   
2200              182               42               118           247  ...   
2202              217               32                12            24  ...   
2203              214               80                30            61  ...   
2204               61                2                 1            21  ...   

      education_Graduation  education_Master  education_PhD  MntTotal  \
0                      3.0               NaN            NaN      1529   
1                      3.0               NaN            NaN        21   
2                      3.0               NaN            NaN       734   
3                      3.0               NaN            NaN        48   
4                      NaN               NaN            5.0       407   
...                    ...               ...            ...       ...   
2198                   3.0               NaN            NaN        19   
2200                   3.0               NaN            NaN      1094   
2202                   3.0               NaN            NaN      1217   
2203                   NaN               4.0            NaN       782   
2204                   NaN               NaN            5.0       151   

      MntRegularProds  AcceptedCmpOverall  marital_status  education_level  \
0                1441                   0          Single       Graduation   
1                  15                   0          Single       Graduation   
2                 692                   0        Together       Graduation   
3                  43                   0        Together       Graduation   
4                 392                   0         Married              PhD   
...               ...                 ...             ...              ...   
2198               16                   0          Single       Graduation   
2200              847                   0         Married       Graduation   
2202             1193                   1        Divorced       Graduation   
2203              721                   0        Together           Master   
2204              130                   0         Married              PhD   

      kids  expenses  
0        0      1529  
1        2        21  
2        0       734  
3        1        48  
4        1       407  
...    ...       ...  
2198     0        19  
2200     1      1094  
2202     0      1217  
2203     1       782  
2204     2       151  

[2021 rows x 43 columns]

### Temos dados nulos nessa base? Será que eles indicam algo? O que fazer com eles? 

In [29]:
df.isnull().sum()

Income                     0
Kidhome                    0
Teenhome                   0
Recency                    0
MntWines                   0
MntFruits                  0
MntMeatProducts            0
MntFishProducts            0
MntSweetProducts           0
MntGoldProds               0
NumDealsPurchases          0
NumWebPurchases            0
NumCatalogPurchases        0
NumStorePurchases          0
NumWebVisitsMonth          0
AcceptedCmp3               0
AcceptedCmp4               0
AcceptedCmp5               0
AcceptedCmp1               0
AcceptedCmp2               0
Complain                   0
Z_CostContact              0
Z_Revenue                  0
Response                   0
Age                        0
Customer_Days              0
marital_Divorced        1975
marital_Married         1351
marital_Single          1728
marital_Together        1637
marital_Widow           2129
education_2n Cycle      2007
education_Basic         2151
education_Graduation    1092
education_Mast

Sim foram encontrados valores nulos nas colunas:

marital_Divorced 

marital_Married 

marital_Single 

marital_Together 

marital_Widow 

education_2n Cycle 

education_Basic 

education_Graduation 

education_Master 

education_PhD 

In [30]:
df['marital_Divorced'].value_counts()

marital_Divorced
1.0    230
Name: count, dtype: int64

Analisando de forma individual uma das colunas, para esse caso usaremos a coluna 'marital_Divorced', e como resposta temos o numero 1 aparecendo 230 vezes , pode se concluir que esse valor numérico representa um sim , ou seja para cada linha que representa um cliente, na coluna em que seu valor faz caracteristica se ele é divorciado ou não , os valores numéricos representa SIM e valores nulos ou valor ausente representao NÃO . 

In [31]:
coluna_nulos = ['marital_Divorced',
'marital_Married',
'marital_Single',
'marital_Together',
'marital_Widow',
'education_2n Cycle',
'education_Basic',
'education_Graduation',
'education_Master',
'education_PhD']

for i in coluna_nulos:
    df[f'bol_'+i] = np.where(df[i].isnull(),0,1)
df

Income  Kidhome  Teenhome  Recency  MntWines  MntFruits  \
0     58138.0        0         0       58       635         88   
1     46344.0        1         1       38        11          1   
2     71613.0        0         0       26       426         49   
3     26646.0        1         0       26        11          4   
4     58293.0        1         0       94       173         43   
...       ...      ...       ...      ...       ...        ...   
2200  61223.0        0         1       46       709         43   
2201  64014.0        2         1       56       406          0   
2202  56981.0        0         0       91       908         48   
2203  69245.0        0         1        8       428         30   
2204  52869.0        1         1       40        84          3   

      MntMeatProducts  MntFishProducts  MntSweetProducts  MntGoldProds  ...  \
0                 546              172                88            88  ...   
1                   6                2                 1             6  ...   
2                 127              111                21            42  ...   
3                  20               10                 3             5  ...   
4                 118               46                27            15  ...   
...               ...              ...               ...           ...  ...   
2200              182               42               118           247  ...   
2201               30                0                 0             8  ...   
2202              217               32                12            24  ...   
2203              214               80                30            61  ...   
2204               61                2                 1            21  ...   

      bol_marital_Divorced  bol_marital_Married  bol_marital_Single  \
0                        0                    0                   1   
1                        0                    0                   1   
2                        0                    0                   0   
3                        0                    0                   0   
4                        0                    1                   0   
...                    ...                  ...                 ...   
2200                     0                    1                   0   
2201                     0                    0                   0   
2202                     1                    0                   0   
2203                     0                    0                   0   
2204                     0                    1                   0   

      bol_marital_Together  bol_marital_Widow  bol_education_2n Cycle  \
0                        0                  0                       0   
1                        0                  0                       0   
2                        1                  0                       0   
3                        1                  0                       0   
4                        0                  0                       0   
...                    ...                ...                     ...   
2200                     0                  0                       0   
2201                     1                  0                       0   
2202                     0                  0                       0   
2203                     1                  0                       0   
2204                     0                  0                       0   

      bol_education_Basic  bol_education_Graduation  bol_education_Master  \
0                       0                         1                     0   
1                       0                         1                     0   
2                       0                         1                     0   
3                       0                         1                     0   
4                       0                         0                     0   
...                   ...                       ...                   ...

### Qual é a média, mediana, 25 percentil, 75 percentil, mínimo e máximo de cada uma das colunas numéricas? 

In [32]:
df.describe()

Income      Kidhome     Teenhome      Recency     MntWines  \
count    2205.000000  2205.000000  2205.000000  2205.000000  2205.000000   
mean    51622.094785     0.442177     0.506576    49.009070   306.164626   
std     20713.063826     0.537132     0.544380    28.932111   337.493839   
min      1730.000000     0.000000     0.000000     0.000000     0.000000   
25%     35196.000000     0.000000     0.000000    24.000000    24.000000   
50%     51287.000000     0.000000     0.000000    49.000000   178.000000   
75%     68281.000000     1.000000     1.000000    74.000000   507.000000   
max    113734.000000     2.000000     2.000000    99.000000  1493.000000   

         MntFruits  MntMeatProducts  MntFishProducts  MntSweetProducts  \
count  2205.000000      2205.000000      2205.000000       2205.000000   
mean     26.403175       165.312018        37.756463         27.128345   
std      39.784484       217.784507        54.824635         41.130468   
min       0.000000         0.000000         0.000000          0.000000   
25%       2.000000        16.000000         3.000000          1.000000   
50%       8.000000        68.000000        12.000000          8.000000   
75%      33.000000       232.000000        50.000000         34.000000   
max     199.000000      1725.000000       259.000000        262.000000   

       MntGoldProds  ...  bol_marital_Divorced  bol_marital_Married  \
count   2205.000000  ...           2205.000000          2205.000000   
mean      44.057143  ...              0.104308             0.387302   
std       51.736211  ...              0.305730             0.487244   
min        0.000000  ...              0.000000             0.000000   
25%        9.000000  ...              0.000000             0.000000   
50%       25.000000  ...              0.000000             0.000000   
75%       56.000000  ...              0.000000             1.000000   
max      321.000000  ...              1.000000             1.000000   

       bol_marital_Single  bol_marital_Together  bol_marital_Widow  \
count         2205.000000           2205.000000        2205.000000   
mean             0.216327              0.257596           0.034467   
std              0.411833              0.437410           0.182467   
min              0.000000              0.000000           0.000000   
25%              0.000000              0.000000           0.000000   
50%              0.000000              0.000000           0.000000   
75%              0.000000              1.000000           0.000000   
max              1.000000              1.000000           1.000000   

       bol_education_2n Cycle  bol_education_Basic  bol_education_Graduation  \
count             2205.000000          2205.000000               2205.000000   
mean                 0.089796             0.024490                  0.504762   
std                  0.285954             0.154599                  0.500091   
min                  0.000000             0.000000                  0.000000   
25%                  0.000000             0.000000                  0.000000   
50%                  0.000000             0.000000                  1.000000   
75%                  0.000000             0.000000                  1.000000   
max                  1.000000             1.000000                  1.000000   

       bol_education_Master  bol_education_PhD  
count           2205.000000        2205.000000  
mean               0.165079           0.215873  
std                0.371336           0.411520  
min                0.000000           0.000000  
25%                0.000000           0.000000  
50%                0.000000           0.000000  
75%                0.000000           0.000000  
max                1.000000           1.000000  

[8 rows x 51 columns]

### Qual é o maior salário encontrado na nossa base? 

In [33]:
salario = df['Income'].max()
print(f'R$ {salario:,}')

R$ 113,734.0


### Qual é a distribuição de salário na nossa base?

In [34]:
def arrumar_txt(txt):
    return f'{txt:,.2f}'

novo_df = (df['Income'].describe().to_frame())

df_format = novo_df['Income'][1:].apply(arrumar_txt).to_frame()

df_format

Income
mean   51,622.09
std    20,713.06
min     1,730.00
25%    35,196.00
50%    51,287.00
75%    68,281.00
max   113,734.00

In [35]:
df_salario = df['Income'].value_counts().to_frame().reset_index()

px.histogram(df_salario,x='Income',y='count',marginal='box')




In [36]:
px.bar(df['education_level'].value_counts())

In [37]:
px.bar(df['marital_status'].value_counts())

In [38]:
df.groupby('marital_status')['kids'].mean().reset_index().sort_values(by='kids',ascending=False)

marital_status      kids
0       Divorced  1.008696
3       Together  0.980634
1        Married  0.967213
4          Widow  0.881579
2         Single  0.859539

In [39]:
df.groupby('marital_status')['kids'].count().reset_index().sort_values(by='kids',ascending=False)

marital_status  kids
1        Married   854
3       Together   568
2         Single   477
0       Divorced   230
4          Widow    76